### PR TITLE
adding PBR Metallic Roughness support for glTF serializer, with unit tests

### PR DESCRIPTION
**Pull request by kcoley**
- Add PBR Metallic Roughness Material support for glTF serializer
- Added Clamp method to Color3 and Color4 class
- Added unit tests

### CODE REVIEW

File: src/Math/babylon.math.ts [hunk 1]
```diff
@@ -278,6 +278,21 @@
...
         }
 
+        /**
+         * Creates a new Color3 with values clamped between min and max.
+         * @param color - original color
+         * @param min - minimum value
+         * @param max - maximum value
+         * @returns - Color3
+         */
+        public static Clamp(color: Color3, min: number = 0, max: number = 1): Color3 {
```
Comment (deltakosh) [inline src/Math/babylon.math.ts]: I would have seen these two functions as non static ones and named them clampToRef perhaps?

Comment (kcoley) [reply src/Math/babylon.math.ts]: Sure I can do that.

File: src/Math/babylon.math.ts [hunk 2]
```diff
@@ -125,6 +125,21 @@
...
 
+        /**
+         * Clamps the rgb values by the min and max values and stores the result into "result".
+         * Returns the unmodified current Color3.
+         * @param min - minimum clamping value.  Defaults to 0
+         * @param max - maximum clamping value.  Defaults to 1
+         * @param result - color to store the result into.
+         * @returns - the original Color3
+         */
+        public clampToRef(min: number = 0, max: number = 1, result: Color3): Color3 {
```
Comment (kcoley) [inline src/Math/babylon.math.ts]: Changed to clampToRef

File: src/Math/babylon.math.ts [hunk 3]
```diff
@@ -388,6 +403,22 @@
...
 
+        /**
+         * Clamps the rgb values by the min and max values and stores the result into "result".
+         * Returns the unmodified current Color4.
+         * @param min - minimum clamping value.  Defaults to 0
+         * @param max - maximum clamping value.  Defaults to 1
+         * @param result - color to store the result into.
+         * @returns - the original Color4
+         */
+        public clampToRef(min: number = 0, max: number = 1, result: Color4): Color4 {
```
Comment (kcoley) [inline src/Math/babylon.math.ts]: Changed to clampToRef

**Pull request merged**